src/gluonts/model/rotbaum/_model.py

```diff
@@ -0,0 +1,277 @@
+        self,
+        model=None,
+        xgboost_params: Optional[dict] = None,
+        clump_size: int = 100,
+    ):
+        """
+        QRX is an algorithm that takes a point estimate algorithm and turns it
+        into a probabilistic forecasting algorithm. By default it uses XGBoost.
+
+        You fit it once, and choose the quantile to predict only at
+        prediction time.
+
+        Prediction is done by taking empirical quantiles of *true values*
+        associated with point estimate predictions close to the point
+        estimate of the given point. The minimal number of associated true
+        values is determined by clump_size.
+
+        The algorithm is (loosely) inspired by quantile regression
+        forests, in that it is predicts quantiles based on associated true
+        values, where the association is based on a point estimate algorithm.
+
+        Parameters
+        ----------
+        model
+            Any point estimate algorithm with .fit and .predict functions.
+        xgboost_params
+            If None, then it uses
+            {"max_depth": 5, "n_jobs": -1, "verbosity": 1,
+             "objective": "reg:squarederror"}
+        clump_size
+            Hyperparameter that determines the minimal size of the list of
+            true values associated with each prediction.
+        """
+        if model:
+            self.model = model
+        else:
+            self.model = self._create_xgboost_model(xgboost_params)
+        self.clump_size = clump_size
+        self.df = None
+        self.processed_df = None
+        self.cell_values = None
+        self.cell_values_dict = None
+
+    @staticmethod
+    def _create_xgboost_model(model_params: Optional[dict] = None):
+        """
+        Creates an xgboost model using specified or default parameters.
+        """
+        if model_params is None:
+            model_params = {
+                "max_depth": 5,
+                "n_jobs": -1,
+                "verbosity": 1,
+                "objective": "reg:squarederror",
+            }
+        return xgboost.sklearn.XGBModel(**model_params)
+
+    @validated()
+    def fit(self, x_train, y_train):
+        """
+        Fits self.model and partitions R^n into cells. More accurately,
+        it creates a dictionary taking predictions on train to lists of
+        associated true values, and puts it in self.cell_values.
+
+        Parameters
+        ----------
+        x_train: list
+            list of lists
+        y_train: list
+        """
+        self.model.fit(x_train, y_train)
+        y_train_pred = self.model.predict(x_train)
+        self.df = pd.DataFrame(
+            {"x": x_train, "y_true": y_train, "y_pred": y_train_pred}
+        )
+        self.cell_values_dict = self.preprocess_df(
+            self.df, clump_size=self.clump_size
+        )
+        self.cell_values = sorted(self.cell_values_dict.keys())
+
+    @staticmethod
+    def clump(dic: Dict, min_num: int) -> Dict:
```

Comment:
Not sure this should be a staticmethod and not just a function.

Also, can you add an example of what this function does?

We use doctests, which automatically verifies that the dostring-examples actually work: https://docs.python.org/3/library/doctest.html

Reply:
Added docstring-examples